builtin/game/register.lua

```diff
@@ -155,9 +192,28 @@ function core.register_item(name, itemdef)
 	getmetatable(itemdef).__newindex = {}
 
 	--core.log("Registering item: " .. itemdef.name)
-	core.registered_items[itemdef.name] = itemdef
-	core.registered_aliases[itemdef.name] = nil
+	core.registered_items[name] = itemdef
+	core.registered_aliases[name] = nil
 	register_item_raw(itemdef)
+
+	if itemdef.can_place then
+		core.override_on_place(name, function(itemstack, placer, pt)
+			local pos = pt.under
+			local node = minetest.get_node(pos)
```

Comment:
This does not look like a nil value to me:
https://github.com/minetest/minetest/blob/master/builtin/game/register.lua#L292